### DISCUSSION TITLE
Flameshot has a desktop transition animation effect when exiting after completing a screenshot. How can I disable this animation effect?

### DISCUSSION THREAD

**Discussion by yzq-del** [flameshot-org/flameshot#3507] [Q&A]
In macOS, Flameshot has a desktop transition animation effect when exiting after completing a screenshot. How can I disable this animation effect?

**Comment by ilyagr** [comment]
I think https://github.com/flameshot-org/flameshot/pull/3625 and https://github.com/flameshot-org/flameshot/issues/2886 are related. So, this may or may not be fixed soon.

**Comment by mmahmoudian** [comment]
That animation has nothing to do with Flameshot. That's how macOS works. You can see the same animation in Finder or any other graphical software too:
1. open a finder
2. make the window fullscreen ( https://support.apple.com/guide/mac-help/use-apps-in-full-screen-mchl9c21d2be/mac )
3. observe the animation
4. close the Finder
5. observe the animation

The #3625 has a big side-effect: "system menu bar not being able to be visible as part of the capture window". For this reason I personally am not so much in favor of it. In general it feels ridiculous to fix an operating system's problem (lack of customizability) inside our software. The only way I can think of to accept the #3625 if to make it configurable so that user can decide which behavior they want, but then that causes another problem: having options specific to each operating system, which I personally really don't like to have. It increases the maintenance burden, makes config files incompatible across OSs, and is confusing for the users.

Your options at the moment are:
1. reduce animations in your mac (I would start with this): https://apple.stackexchange.com/a/293926/105984
2. specifically turn off animation for each program (it is a hacky way, but with minimum side-effects and is easy to do): https://apple.stackexchange.com/a/146951/105984
3. disable animations in macOS (very hacky, I personally would not recommend it, but it is your computer and you should decide): https://apple.stackexchange.com/questions/14001/how-to-turn-off-all-animations-on-os-x
4. last but not the least, a very obvious and least helpful approach: stop using mac ¯\_(ツ)_/¯

**Reply by ilyagr** [reply]
I initially thought that Flameshot somehow suppressed the animation when opening the full-screen window, but I guess it doesn't, that animation is simply less annoying (it only involves the menu bar). That's unfortunate, it gave me hope that it could suppress the "exiting full screen" animation as well.

> 2. specifically turn off animation for each program (it is a hacky way, but with minimum side-effects and is easy to do): https://apple.stackexchange.com/a/146951/105984

I tried adding

```
    <key>LSUIPresentationMode</key>
    <integer>4</integer>
```

to Flameshot's `Info.plist`, and it didn't seem to help. There were weird changes -- Flameshot started showing in the dock and didn't show a notification anymore and, less surprisingly, I did have to re-gift Flameshot with screen recording permissions -- but the annoying full-screen animation when exiting Flameshot remained. Have you had better luck? (I'm on Sonoma 14.5 on Apple Silicon, for reference).

**Reply by ilyagr** [reply]
> 1. reduce animations in your mac (I would start with this): https://apple.stackexchange.com/a/293926/105984

I tired this, and it does help a bit. It's pretty subjective, the new animation is *differently* annoying, but it feels slightly more intuitive and thus less jarring when exiting the screenshotting interface.

Thanks for the advice!

**Comment by igorgatis** [comment]
@yzq-del try:

<img width="790" height="286" alt="image" src="https://github.com/user-attachments/assets/65f5e4e1-d039-44cb-8f25-18ecaf022742" />

Ref: https://support.apple.com/en-sa/guide/mac-help/mchlc03f57a1/mac
